zetasql/resolved_ast/query_expression.cc

```diff
@@ -337,6 +347,11 @@ bool QueryExpression::CanSetLimitClause() const {
 }
 bool QueryExpression::CanSetOffsetClause() const { return !HasOffsetClause(); }
 
+bool QueryExpression::CanSetTopClause() const {
+  return HasSelectClause() && !HasTopClause() && !HasLimitClause() &&
```

Comment:
I don't see `HasSelectClause()` condition in `CanSetLimitClause`. Why do we need it?